lib/composite_primary_keys/composite_predicates.rb

```diff
@@ -52,40 +52,55 @@ def cpk_join_predicate(table1, key1, table2, key2)
 
     def cpk_in_predicate(table, primary_keys, ids)
       if primary_keys.length == 2
-        low_cardinality_key_part, high_cardinality_key_part = if ids.map(&:second).uniq.size > ids.map(&:first).uniq.size
-                                                                %i[first second]
-                                                              else
-                                                                %i[second first]
-                                                              end
-
-        groups = ids.group_by(&low_cardinality_key_part).transform_values do |values|
-          values.map(&high_cardinality_key_part)
+        cpk_in_predicate_with_grouped_keys(table, primary_keys, ids)
+      else
+        cpk_in_predicate_with_non_grouped_keys(table, primary_keys, ids)
+      end
+    end
+
+    def cpk_in_predicate_with_non_grouped_keys(table, primary_keys, ids)
+      and_predicates = ids.map do |id|
+        cpk_id_predicate(table, primary_keys, id)
+      end
+
+      cpk_or_predicate(and_predicates)
+    end
+
+    def cpk_in_predicate_with_grouped_keys(table, primary_keys, ids)
+      keys_by_first_key_part = Hash.new { |h, k| h[k] = [] }
```

Comment:
Nitpick. Can this be Hash.new { |hash, key| hash[key] = [] }?

Reply:
Done.